### PR TITLE
[T2A2][T16-A3]Simaria Yesha Bharat

### PR DESCRIPTION
**Pull request by yeshasimaria**

### CODE REVIEW

**Comment by yeshasimaria** [comment]
Ready for review.

File: src/seedu/addressbook/AddressBook.java [hunk 1]
```diff
@@ -794,11 +794,11 @@ private static void addPersonToAddressBook(String[] person) {
...
      */
     private static boolean deletePersonFromAddressBook(String[] exactPerson) {
-        final boolean changed = ALL_PERSONS.remove(exactPerson);
-        if (changed) {
+        final boolean isChanged = ALL_PERSONS.remove(exactPerson);
+        if (isChanged) {
             savePersonsToFile(getAllPersonsInAddressBook(), storageFilePath);
         }
-        return changed;
+        return isChanged;
```
Comment (sooyj) [inline src/seedu/addressbook/AddressBook.java:801]: Good: Boolean variable naming convention.

File: src/seedu/addressbook/AddressBook.java [hunk 2]
```diff
@@ -1012,12 +1012,12 @@ private static String extractEmailFromPersonString(String encoded) {
...
         // email is last arg, target is from prefix to end of string
         if (indexOfEmailPrefix > indexOfPhonePrefix) {
-            return removePrefixSign(encoded.substring(indexOfEmailPrefix, encoded.length()).trim(),
+            return removePrefix(encoded.substring(indexOfEmailPrefix, encoded.length()).trim(),
                     PERSON_DATA_PREFIX_EMAIL);
 
         // email is middle arg, target is from own prefix to next prefix
         } else {
-            return removePrefixSign(
+            return removePrefix(
```
Comment (sooyj) [inline src/seedu/addressbook/AddressBook.java:1020]: Good change of name to remove ambiguous term "sign".

File: src/seedu/addressbook/AddressBook.java [hunk 3]
```diff
@@ -1144,14 +1144,14 @@ private static String getUsageInfoForExitCommand() {
      */
 
     /**
-     * Removes sign(p/, d/, etc) from parameter string
+     * Removes first occurrence of prefix from parameter string
      *
-     * @param s  Parameter as a string
-     * @param sign  Parameter sign to be removed
+     * @param fullString  Parameter as a string
+     * @param prefix  Parameter sign to be removed
```
Comment (sooyj) [inline src/seedu/addressbook/AddressBook.java:1150]: Probably should remove the mention of "sign" in this comment?

File: src/seedu/addressbook/AddressBook.java [hunk 4]
```diff
@@ -1144,14 +1144,14 @@ private static String getUsageInfoForExitCommand() {
...
      * @return  string without the sign
      */
-    private static String removePrefixSign(String s, String sign) {
-        return s.replace(sign, "");
+    private static String removePrefix(String fullString, String prefix) {
+        return fullString.replaceFirst(prefix, "");
```
Comment (sooyj) [inline src/seedu/addressbook/AddressBook.java:1154]: Correct change of code logic.

**Review by sooyj — COMMENTED** [review]
Overall a good attempt :-)

**Pull request closed without merging**